includes/class-wc-install.php

```diff
@@ -565,7 +569,21 @@ private static function create_tables() {
 
 		// Add constraint to download logs if the columns matches.
 		if ( ! empty( $download_permissions_column_type ) && ! empty( $download_log_column_type ) && $download_permissions_column_type === $download_log_column_type ) {
-			$wpdb->query( "ALTER TABLE {$wpdb->prefix}wc_download_log ADD FOREIGN KEY (permission_id) REFERENCES {$wpdb->prefix}woocommerce_downloadable_product_permissions(permission_id) ON DELETE CASCADE" );
+			$fk_result = $wpdb->get_row( "
+				SELECT COUNT(*) AS fk_count
+				FROM information_schema.TABLE_CONSTRAINTS
+				WHERE CONSTRAINT_SCHEMA = '{$wpdb->dbname}'
+				AND CONSTRAINT_NAME = 'fk_wc_download_log_permission_id'
```

Comment:
I would suggest we limit the query using `AND TABLE_NAME = '{$wpdb->prefix}wc_download_log'`. I say this because without specifying a table name MySQL will scan all databases and this is potentially slow for servers having a significant number of databases. Unless there is a reason not to specify the table name?

```
MariaDB [wccore]> explain SELECT * FROM information_schema.TABLE_CONSTRAINTS WHERE CONSTRAINT_SCHEMA = 'wccore' AND CONSTRAINT_NAME = 'fk_wc_download_log_permission_id' AND CONSTRAINT_TYPE = 'FOREIGN KEY';
+------+-------------+-------------------+------+---------------+------+---------+------+------+-----------------------------------------------------+
| id   | select_type | table             | type | possible_keys | key  | key_len | ref  | rows | Extra                                               |
+------+-------------+-------------------+------+---------------+------+---------+------+------+-----------------------------------------------------+
|    1 | SIMPLE      | TABLE_CONSTRAINTS | ALL  | NULL          | NULL | NULL    | NULL | NULL | Using where; Open_full_table; Scanned all databases |
+------+-------------+-------------------+------+---------------+------+---------+------+------+-----------------------------------------------------+
1 row in set (0.00 sec)
MariaDB [wccore]> explain SELECT COUNT(*) AS fk_count FROM information_schema.TABLE_CONSTRAINTS WHERE CONSTRAINT_SCHEMA = 'wccore' AND TABLE_NAME = 'wp_wc_download_log' AND CONSTRAINT_NAME = 'fk_wc_download_log_permission_id' AND CONSTRAINT_TYPE = 'FOREIGN KEY';
+------+-------------+-------------------+------+---------------+------------+---------+------+------+--------------------------------------------------+
| id   | select_type | table             | type | possible_keys | key        | key_len | ref  | rows | Extra                                            |
+------+-------------+-------------------+------+---------------+------------+---------+------+------+--------------------------------------------------+
|    1 | SIMPLE      | TABLE_CONSTRAINTS | ALL  | NULL          | TABLE_NAME | NULL    | NULL | NULL | Using where; Open_full_table; Scanned 1 database |
+------+-------------+-------------------+------+---------------+------------+---------+------+------+--------------------------------------------------+
1 row in set (0.00 sec)
```

I'm sorry I didn't notice this in my first review.